### PR TITLE
[com_weblinks] Make ACL core.edit.own work (PR for 11466)

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/11466 (com_weblinks part).

#### Summary of Changes

The ACL core.edit.own is not working in com_weblinks.
This PR makes it work.

#### Testing Instructions

1. Use latest staging
2. Besides the Super User, create a user "test" added to "Administrator" group
3. Go to com_weblinks Options, Permissions tab and diable "Edit" for "Administrator" group
4. Now create a new weblink with the "test" user (use another browser or a private window)
5. Try to edit that item. You can't edit our own. **Bug**
6. Now do the same test but with a weblink category Permission. You can't edit our own. **Bug**
7. Apply patch, repeat step 5. and 6. and now you can  edit our own items.
8. Code review

Also use the two users to do a general test with the com_weblinks edit permissions to confirm all is fine.

#### Documentation Changes Required

None.

#### Notes

This problem was discovered in GsoC 2016 Multilingual project.

This is very similiar with https://github.com/joomla/joomla-cms/pull/11502 (com_newsfeeds) and https://github.com/joomla/joomla-cms/pull/11503 (com_contact).

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
i guess the test errors have nothing to do with this PR ...

File: src/administrator/components/com_weblinks/controllers/weblink.php [hunk 1]
```diff
@@ -60,21 +60,29 @@ protected function allowAdd($data = array())
...
 
-		// Since there is no asset tracking, revert to the component permissions.
-		return parent::allowEdit($data, $key);
+		$user = JFactory::getUser();
+
+		// Check if can edit own core.edit.own.
+		$canEditOwn = $user->authorise('core.edit.own', $this->option . '.category.' . (int) $item->catid) && $item->created_by == $user->id;
+
+		// Check the category core.edit permissions.
+		return $canEditOwn || $user->authorise('core.edit', $this->option . '.category.' . (int) $item->catid);
```
Comment (yvesh) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: Can you save that in a variable before? That looks really not nice ;-)

Comment (andrepereiradasilva) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: what variable? you want a $canEdit ? is it really needed .... ?

Comment (yvesh) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: Just my opinion :) That return statement is not good readable imo.

**Comment by yvesh** [comment]
@andrepereiradasilva i get an SQL syntax error when i apply your patch. System tests also fail because of that

screenshot 2016-08-07 09 43 14

**Comment by yvesh** [comment]
You are missing a , after a.created_by

File: src/administrator/components/com_weblinks/models/weblinks.php [hunk 2]
```diff
@@ -132,7 +132,7 @@ protected function getListQuery()
 		$query->select(
 			$this->getState(
 				'list.select',
-				'a.id, a.title, a.alias, a.checked_out, a.checked_out_time, a.catid,' .
+				'a.id, a.title, a.alias, a.checked_out, a.checked_out_time, a.catid, a.created_by ' .
```
Comment (yvesh) [inline src/administrator/components/com_weblinks/models/weblinks.php]: Comma missing after a.created_by

Comment (andrepereiradasilva) [inline src/administrator/components/com_weblinks/models/weblinks.php]: ups sorry will correct

File: src/administrator/components/com_weblinks/controllers/weblink.php (hunk 1 above)
Comment (andrepereiradasilva) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: the idea is to only check authorize to core.edit if user cannot edit own.

**Comment by alikon** [comment]
i've tested successfully

**Comment by infograf768** [comment]
OK here

**Comment by yvesh** [comment]
@alikon @infograf768 Thanks for testing, please use the issue tracker next time like on core :-)

**Comment by yvesh** [comment]
I have tested this item :white_check_mark: successfully on cfd842f5e48b6cd1b7cc58d7e130cdffadcd2f3d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/weblinks/255">issues.joomla.org/weblinks/255</a>.</sub>

File: src/administrator/components/com_weblinks/controllers/weblink.php (hunk 1 above)
Comment (yvesh) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: That's fine, but you could also save that to a variable ;-)

```php
$canEdit = $canEditOwn || $user->..

return $canEdit; 
```

But as i said that's just my opinion :-)

Comment (mbabker) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: Personally I'd say don't change it.  Unless it's going to improve readability or going to be reused I try to not create variables needlessly.  This is a case where it's doing neither IMO.  Of course, coding styles and opinions have a lot of things in common 😉

Comment (andrepereiradasilva) [inline src/administrator/components/com_weblinks/controllers/weblink.php]: i'm think like @mbabker on this. if we don't use a variable i don't like to create it.
but yes is more a matter of coding style.

**Comment by infograf768** [comment]
Didn't even know we have a specific issue tracker for such project. 
Found it: https://issues.joomla.org/tracker/weblinks

**Comment by andrepereiradasilva** [comment]
will this be merged?

And BTW when there will be a new weblinks release?

**Comment by yvesh** [comment]
// cc @chrisdavenport @rdeutz

**Pull request merged**